opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp

```diff
@@ -81,6 +99,57 @@ namespace Opm {
                 for (size_t globalIndex = 0; globalIndex < multpvData.size(); globalIndex++)
                     values[globalIndex] *= multpvData[globalIndex];
             }
+
+            // deal with the region multiplier for porosity
+            if (deck->hasKeyword("MULTREGP")) {
+                const DeckKeyword& multregpKeyword = deck->getKeyword("MULTREGP");
+                for (int recordIdx = 0; recordIdx < multregpKeyword.size(); ++recordIdx) {
+                    const DeckRecord& multregpRecord = multregpKeyword.getRecord(recordIdx);
+
+                    int regionIdx = multregpRecord.getItem("REGION").template get<int>(0);
+                    std::string regionType = multregpRecord.getItem("REGION_TYPE").template get<std::string>(0);
+                    double multValue = multregpRecord.getItem("MULTIPLIER").template get<double>(0);
+                    boost::to_upper(regionType);
```

Comment:
> You could check the MULTREGT code, I am quite certain there already is code for this minor detail there.

sure: for MULTREGT, the current implementation does not seem to do anything:

https://github.com/OPM/opm-parser/blob/master/opm/parser/eclipse/EclipseState/Eclipse3DProperties.cpp#L475
and https://github.com/OPM/opm-parser/blob/master/opm/parser/eclipse/EclipseState/Grid/MULTREGTScanner.cpp#L36

which corresponds to the documented behaviour. I'm just wondering whether that's what the reference implementation does, because there are other cases that are quite similar IMO where the documentation deviates from its implementation. (lower-case keywords anyone?)

I suppose you're right that if MULTREGP needs to be fixed, the MULTREGT code probably needs changes, too.